zetaclient/chains/base/confirmation.go

```diff
@@ -0,0 +1,85 @@
+package base
+
+// CalcUnscannedBlockRangeInboundSafe calculates the unscanned block range using inbound safe confirmation count.
+// It returns a range of blocks [from, end (exclusive)) that need to be scanned.
+func (ob *Observer) CalcUnscannedBlockRangeInboundSafe(blockLimit uint64) (from uint64, end uint64) {
+	lastBlock := ob.LastBlock()
+	lastScanned := ob.LastBlockScanned()
+	confirmation := ob.ChainParams().InboundConfirmationSafe()
+
+	return calcUnscannedBlockRange(lastBlock, lastScanned, confirmation, blockLimit)
+}
+
+// CalcUnscannedBlockRangeInboundFast calculates the unscanned block range using inbound fast confirmation count.
+// It returns a range of blocks [from, end (exclusive)) that need to be scanned.
+func (ob *Observer) CalcUnscannedBlockRangeInboundFast(blockLimit uint64) (from uint64, end uint64) {
```

Comment:
WDYT about this naming? IMO it's clearer
```suggestion
func (ob *Observer) GetScanRangeInboundFast(blockLimit uint64) (from uint64, end uint64) {
```

Reply:
https://github.com/zeta-chain/node/pull/3496/commits/03f6bcc4e8f62b6facf287077753cede38ef6900